If app with emoji name is installed, typing into search bar crashes

On my phone, KISS's AppProvider is crashing whenever I type into the search bar. Uninstalling Image Analysis Toolset, an app with emoji in the app-drawer name (but not the Settings name), then KISS Settings/Advanced/Restart fixes this crash.

```
java.lang.RuntimeException: An error occurred while executing doInBackground()
	at android.os.AsyncTask$3.done(AsyncTask.java:318)
	at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:354)
	at java.util.concurrent.FutureTask.setException(FutureTask.java:223)
	at java.util.concurrent.FutureTask.run(FutureTask.java:242)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607)
	at java.lang.Thread.run(Thread.java:761)
Caused by: java.lang.StringIndexOutOfBoundsException: String index out of range: 9
	at java.lang.String.codePointAt(String.java:602)
	at fr.neamar.kiss.dataprovider.AppProvider.getResults(AppProvider.java:169)
	at fr.neamar.kiss.searcher.QuerySearcher.doInBackground(QuerySearcher.java:1263)
	at android.os.AsyncTask$2.call(AsyncTask.java:304)
	at java.util.concurrent.FutureTask.run(FutureTask.java:237)
	... 3 more
```

I discovered that the "Scoop" Xposed module automatically captures stack traces into your notifications. It's really helpful and I would recommend it for everyone.

--- a/app/src/main/java/fr/neamar/kiss/normalizer/StringNormalizer.java
+++ b/app/src/main/java/fr/neamar/kiss/normalizer/StringNormalizer.java
@@ -66,8 +66,12 @@ public class StringNormalizer {
                         break;
 
                     default:
-                        resultString.appendCodePoint(Character.toLowerCase(resultChar));
-                        resultMap.add(inputOffset);
+                        int mapIdx = resultString.length();
+                        int lwrCodePoint = Character.toLowerCase(resultChar);
+                        resultString.appendCodePoint( lwrCodePoint );
+                        if ( Character.charCount( lwrCodePoint ) == 2 )
+                            resultMap.add( mapIdx );
+                        resultMap.add( mapIdx );
                 }
 
                 decomposedCharOffset += Character.charCount(resultChar);

